10_rest/runnerdave/pkg/puppy/rest_test.go

```diff
@@ -0,0 +1,227 @@
+package puppy_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	chi "github.com/go-chi/chi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	puppy "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy"
+	store "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy/store"
+)
+
+var (
+	rs puppy.RestStorer
+)
+
+func startAPIHandler() *chi.Mux {
+	r := chi.NewRouter()
+	rs = puppy.RestStorer{store.NewSyncStore()}
+	newPuppy := puppy.Puppy{
+		Breed:  "ULTRASURE",
+		Colour: "green",
+		Value:  2732.81,
+	}
+	if err := rs.Db.CreatePuppy(newPuppy); err != nil {
+		panic(err)
+	}
+
+	r.Get(rs.GetPuppyRoute(), rs.GetPuppy)
+	r.Post(rs.PostPuppyRoute(), rs.CreatePuppy)
+	r.Put(rs.PutPuppyRoute(), rs.UpdatePuppy)
+	r.Delete(rs.DeletePuppyRoute(), rs.DeletePuppy)
```

Comment:
I think it'd rewrite this as something like:
I also think it doesn't belong with the test code but the rest.go file.
```go
	r.Route("/api/puppy", func(r chi.Router) {
		r.Post("/", rs.CreatePuppy)
	        r.Route("/{id}", func(r chi.Router) {
		     r.Use(CtxIDMiddleware) // see chi docs for sample, e.g. REST Preview: https://github.com/go-chi/chi#examples
		     r.Put("/", h.handlePut)
		     r.Delete("/", h.handleDelete)
		     r.Get("/history", h.handleGetHistory)
	    })
	})
```

Reply:
very useful comment, thanks a lot, implemented